OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -126,6 +126,37 @@ List<IFinalizedRenderable> GenerateRenderables()
 			return renderables;
 		}
 
+		List<IFinalizedRenderable> GenerateOverlayRenderables(HashSet<Actor> actorsInBox)
+		{
+			var aboveShroud = World.ActorsWithTrait<IRenderAboveShroud>()
+				.Where(a => a.Actor.IsInWorld && !a.Actor.Disposed && (!a.Trait.SpatiallyPartitionable || actorsInBox.Contains(a.Actor)))
+					.SelectMany(a => a.Trait.RenderAboveShroud(a.Actor, this));
+
+			var aboveShroudSelected = World.ActorsWithTrait<IRenderAboveShroudWhenSelected>()
```

Comment:
Most actors have at least one `IRenderAboveShroudWhenSelected` trait, so this change introduces a lot more work.  Can we please keep this enumerating over the much smaller `World.Selection.Actors.SelectMany()` collection?